cups-filters: broken build from poppler_utils update

## Issue description
cups-filters fails to build on recent master (333a6646a8b9475bce8905c76855d2aa4d0a7411)

Reverting the recent poppler_utils update (7d016502d88a89bebb2fd8728a5bb40702a7feba) resolves the problem, and cups-filters builds properly again. This was PR #43915. 

### Steps to reproduce
Failure: `nix-build -A cups-filters`
```
  CC       gstoraster-gstoraster.o
  CC       mupdftoraster-mupdftoraster.o
  CC       foomatic_rip-foomaticrip.o
  CC       foomatic_rip-options.o
  CC       foomatic_rip-pdf.o
  CC       foomatic_rip-postscript.o
  CC       foomatic_rip-process.o
  CC       foomatic_rip-renderer.o
filter/pdf.cxx: In function 'int pdf_fill_form(pdf_t*, opt_t*)':
filter/pdf.cxx:740:44: error: invalid conversion from 'const GooString*' to 'GooString*' [-fpermissive]
             field_name = ff->getPartialName();
                          ~~~~~~~~~~~~~~~~~~^~
make[1]: *** [Makefile:3937: bannertopdf-pdf.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/cups-filters-1.20.4'
make: *** [Makefile:1975: all] Error 2
builder for '/nix/store/bshxw2d6zg5cw134nv5zch9favrn7gqp-cups-filters-1.20.4.drv' failed with exit code 2
error: build of '/nix/store/bshxw2d6zg5cw134nv5zch9favrn7gqp-cups-filters-1.20.4.drv' failed
```
Success: `git revert 7d016502d88a89bebb2fd8728a5bb40702a7feba && nix-build -A cups_filters`
## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.58, NixOS, 18.09.git.f134990 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-17.09pre104936.b0d07aa894"`
 - channels(calvertvl): `""`
 - nixpkgs: `/home/calvertvl/repos/nixpkgs`

--- a/pkgs/misc/cups/filters.nix
+++ b/pkgs/misc/cups/filters.nix
@@ -2,6 +2,7 @@
 , libjpeg, libpng, perl, ijs, qpdf, dbus, avahi
 , makeWrapper, coreutils, gnused, bc, gawk, gnugrep, which, ghostscript
 , mupdf
+, fetchpatch
 }:
 
 let
@@ -15,6 +16,16 @@ in stdenv.mkDerivation rec {
     url = "https://openprinting.org/download/cups-filters/${name}.tar.xz";
     sha256 = "0sjkmclcb1r77015wllsyz26272br3s17v6b1q2xwb2nm2gnwx9k";
   };
+
+  patches = [
+    # This patch fixes cups-filters when compiled with poppler-0.67.0.
+    # Issue: https://github.com/OpenPrinting/cups-filters/pull/50
+    # PR: https://github.com/OpenPrinting/cups-filters/pull/51
+    (fetchpatch {
+      url = "https://github.com/OpenPrinting/cups-filters/commit/219de01c61f3b1ec146abf142d0dfc8c560cc58e.patch";
+      sha256 = "0f0lql3rbm2g8mxrpigfyi8fb4i2g4av20g417jzdilp60jq0ny8";
+    })
+  ];
 
   nativeBuildInputs = [ pkgconfig makeWrapper ];
 

